### PR TITLE
Add API for get single merge  request, create note on merge request, get pipelines for merge request, delete for webhooks

### PR DESCRIPTION
**Pull request by saber-wang**
Get single MR And Add MergeRequestDetail
Add List MR pipelines
Create Merge Request note request

### CODE REVIEW

**Comment by jetersen** [comment]
@saber-wang test failed: https://github.com/nmklotas/GitLabApiClient/pull/137/checks?check_run_id=585099995#step:9:90

What prevents on from extending `MergeRequest` to have the additional info.

Yes some of it might be empty when requesting multiple MRs, however I do not see a harm in that.

**Comment by jetersen** [comment]
no need to close the PR, you can just push your changes to the branch

**Comment by saber-wang** [comment]
> test failed: https://github.com/nmklotas/GitLabApiClient/pull/137/checks?check_run_id=585099995#step:9:90

I don't know why the test failed


> What prevents on from extending `MergeRequest` to have the additional info.
> 
> Yes some of it might be empty when requesting multiple MRs, however I do not see a harm in that.

I'm just worried about misdirection

**Comment by jetersen** [comment]
@saber-wang the test failure is pretty clear 🤔
```
[xUnit.net 00:03:32.31]       System.InvalidCastException : Unable to cast object of type 'GitLabApiClient.Models.MergeRequests.Responses.MergeRequest' to type 'GitLabApiClient.Models.MergeRequests.Responses.MergeRequestDetail'.
[xUnit.net 00:03:32.31]       Stack Trace:
[xUnit.net 00:03:32.31]            at FluentAssertions.Primitives.ReferenceTypeAssertions`2.Match[T](Expression`1 predicate, String because, Object[] becauseArgs)
[xUnit.net 00:03:32.31]         /home/runner/work/GitLabApiClient/GitLabApiClient/test/GitLabApiClient.Test/MergeRequestClientTest.cs(38,0): at GitLabApiClient.Test.MergeRequestClientTest.CreatedMergeRequestCanBeRetrieved()
```

File: src/GitLabApiClient/PipelineClient.cs [hunk 1]
```diff
@@ -43,7 +43,7 @@ public async Task<IList<Job>> GetJobsAsync(ProjectId projectId, int pipelineId,
             var queryOptions = new JobQueryOptions();
             options?.Invoke(queryOptions);
 
-            var url = _jobQueryBuilder.Build($"projects/{projectId}/pipelines/{pipelineId}/jobs", queryOptions);
+            string url = _jobQueryBuilder.Build($"projects/{projectId}/pipelines/{pipelineId}/jobs", queryOptions);
```
Comment (jetersen) [inline src/GitLabApiClient/PipelineClient.cs:46]: Please revert
```suggestion
            var url = _jobQueryBuilder.Build($"projects/{projectId}/pipelines/{pipelineId}/jobs", queryOptions);
```

File: src/GitLabApiClient/Models/PageQuery.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+using System.Text;
+
+namespace GitLabApiClient.Models
+{
+    public class PageQuery
+    {
+        public int? Page { get; set; }
+        public int? Per_Page { get; set; }
+    }
+}
```
Comment (jetersen) [inline src/GitLabApiClient/Models/PageQuery.cs:12]: This seems unused

Comment (saber-wang) [reply src/GitLabApiClient/Models/PageQuery.cs:12]: I forgot to delete it. I'll fix it

**Pull request merged**